oci/private/image.sh.tpl

```diff
@@ -131,6 +135,11 @@ for ARG in "$@"; do
             FIXED_ARGS+=("--entrypoint=$in")
           done <"${ARG#--entrypoint-file=}"
           ;;
+        (--flatten=*)
```

Comment:
I am torn about this.

One the one hand, this change doesn't add complexity to rules_oci it seems, just more faithfully presents the underling CLI of the tool we wrap.

OTOH I think this will result in users having massive outputs from actions, and then having to take special care that these aren't fetched back and forth from a remote cache to a local bazel server. I think the lack of a flattening ability can actually be considered a feature of rules_oci, that you can't recreate the bad behavior you get under rules_docker.

I think we need to find a way to solve the underlying problem (very slow load into docker daemon) either without needing flattening, or by putting some guardrails and warnings on this feature so that users understand the risks.

Reply:
> I am torn about this.

Yeah me too. I left it in draft because I don't have enough justification to put it up for review yet

> I think we need to find a way to solve the underlying problem (very slow load into docker daemon)

I profiled `docker buildx --builder=container build --load - < Dockerfile.localstack` on my local machine OSX/amd64

`cat Dockerfile.localstack`:
```
FROM localstack/localstack:2.2.0
```

`docker --debug buildx --builder=container inspect`:
```
Name:          container
Driver:        docker-container
Last Activity: 2023-10-09 01:09:41 +0000 UTC

Nodes:
Name:           container0
Endpoint:       desktop-linux
Driver Options: env.JAEGER_TRACE="localhost:6831" network="host"
Status:         running
Buildkit:       v0.12.2
Platforms:      linux/amd64, linux/amd64/v2, linux/amd64/v3, linux/arm64, linux/riscv64, linux/ppc64le, linux/s390x, linux/386, linux/mips64le, linux/mips64, linux/arm/v7, linux/arm/v6
Features:
 Cache export:       true
 Docker exporter:    true
 Multiple platforms: true
 OCI exporter:       true
Labels:
 org.mobyproject.buildkit.worker.executor:         oci
 org.mobyproject.buildkit.worker.hostname:         docker-desktop
 org.mobyproject.buildkit.worker.network:          host
 org.mobyproject.buildkit.worker.oci.process-mode: sandbox
 org.mobyproject.buildkit.worker.selinux.enabled:  false
 org.mobyproject.buildkit.worker.snapshotter:      overlayfs
GC Policy rule#0:
 All:           false
 Filters:       type==source.local,type==exec.cachemount,type==source.git.checkout
 Keep Duration: 48h0m0s
 Keep Bytes:    488.3MiB
GC Policy rule#1:
 All:           false
 Keep Duration: 1440h0m0s
 Keep Bytes:    27.94GiB
GC Policy rule#2:
 All:        false
 Keep Bytes: 27.94GiB
GC Policy rule#3:
 All:        true
 Keep Bytes: 27.94GiB
```
Trace in Jaeger (GitHub won't let me attach the trace .json):
<img width="1680" alt="Screenshot 2023-10-09 at 9 53 45 AM" src="https://github.com/bazel-contrib/rules_oci/assets/555944/1498cdb5-69d1-4008-8cf7-fb9ca8477df4">

The only thing that sticks out is the number of logs that happen almost simultaneously:
<img width="1081" alt="Screenshot 2023-10-09 at 10 08 36 AM" src="https://github.com/bazel-contrib/rules_oci/assets/555944/238e0252-08c9-4472-8b4e-0cc202c4e4ed">

These logs start at 5.37s and end at 5.42s. So it's like the actual streaming happened really fast, but whatever happens after that is slow and unmeasured?

Anyway, like I said I'm not advocating for merging this PR. At least not until I understand what the slow part actually is

Reply:
My take on this PR is that this can be done in genrule, and generally there's little to no benefit for rules_oci to maintain this long term. A flattening rule like this will put a significant strain on the remote cache when there is more than one oci_image that flattens. 

It don't think we can accept this contribution.